aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,769 @@
+                shape=sample_chunk_shape,
+                dtype=sample.dtype,
+            )
+
+        # Convert the numpy array of lazy readers into a dask array and fill the inner-most empty dimensions with chunks
+        merged = da.block(lazy_arrays.tolist())
+
+        # Because we have set certain dimensions to be chunked and others not
+        # we will need to transpose back to original dimension ordering
+        # Example being, if the original dimension ordering was "SZYX" and we want to chunk by "S", "Y", and "X"
+        # We created an array with dimensions ordering "ZSYX"
+        transpose_indices = []
+        transpose_required = False
+        for i, d in enumerate(Dimensions.DefaultOrder):
+            new_index = blocked_dimension_order.index(d)
+            if new_index != i:
+                transpose_required = True
+                transpose_indices.append(new_index)
+            else:
+                transpose_indices.append(i)
+
+        # Only run if the transpose is actually required
+        # The default case is "Z", "Y", "X", which _usually_ doesn't need to be transposed because that is _usually_
+        # The normal dimension order of the LIF file anyway
+        if transpose_required:
+            merged = da.transpose(merged, tuple(transpose_indices))
+
+        # Because dimensions outside of Y and X can be in any order and present or not
+        # we also return the dimension order string.
+        return merged, "".join(dims)
+
+    @property
+    def dask_data(self) -> da.core.Array:
+        """
+        Returns
+        -------
+        Constructed dask array where each chunk is a delayed read from the LIF file.
+        Places dimensions in the native order (i.e. "TZCYX")
+        """
+        if self._dask_data is None:
+            self._dask_data, self._dims = LifReader._daread(
+                self._file,
+                self._chunk_offsets,
+                self._chunk_lengths,
+                chunk_by_dims=self.chunk_by_dims,
+                S=self.specific_s_index
+            )
+
+        return self._dask_data
+
+    def dtype(self) -> np.dtype:
+        """
+        The data type of the underlying numpy ndarray, ie uint8, uint16, uint32 etc.
+
+        Returns
+        -------
+        numpy.dtype
+            The data format used to store the data in the Leica lif file and the read numpy.ndarray.
+        """
+        return self.dask_data.dtype
+
+    @property
+    def metadata(self) -> Element:
+        """
+        Load and return the metadata from the LIF file
+
+        Returns
+        -------
+        The lxml Element Tree of the metadata
+        """
+        # We can't serialize lxml element trees so don't save the tree to the object state
+        meta_xml, header = utilities.get_xml(self._file)
+        return meta_xml
+
+    def _size_of_dimension(self, dim: str) -> int:
+        if dim in self.dims:
+            return self.dask_data.shape[self.dims.index(dim)]
```

Comment:
I has to construct the dask array but doesn't have to do any reading.

Reply:
just wondering if these dims could be got from a (potentially cheaper?) metadata read instead.

Reply:
It's possible I'm could be wrong but it doesn't hit me as overly intensive it's just using the metadata dims to create dask.array and returning the shape. I borrowed it from one of the other readers because it seemed like a nice consistent way to leverage other code.